packages/iov-bns/src/bnsconnection.ts

```diff
@@ -613,15 +614,12 @@ export class BnsConnection implements BcpAtomicSwapConnection {
       throw new Error("Received transaction of unsupported kind.");
     }
     // get gconf fee
-    // TODO: get product fee for this type
+    const min = await this.getMinimumFee();
+    // get product fee for this type
+    const product = await this.getProductFee(transaction.kind);
     // take maximum
-    const res = await this.query("/", Encoding.toAscii("gconf:cash:minimal_fee"));
-    if (res.results.length !== 1) {
-      throw new Error("Received unexpected number of fees");
-    }
-    const data = Encoding.fromAscii(res.results[0].value);
-    const amount = decodeJsonAmount(data);
-    return { tokens: amount };
+    const fee = product || min;
```

Comment:
I can see the elegance of this code, but in this case we should save the `getMinimumFee` call when `getProductFee` returns something.

Reply:
This implementation also means that the "min fee" is not a minimum fee but a default fee. In the case procuct fee < min fee, the fee is less than min fee.